### ISSUE TITLE
Handling of out of scope messages

### ISSUE THREAD

**Issue by buivietan** [RasaHQ/rasa_nlu#1875]
Hi there, 
I would like to ask if we can handle the response of the bot if the user input doesn't belong to the list of the intents. For example in restaurant bot if user would ask "I would like to book the flight" the bot can response something like "Sorry I don't understand, can you repeat it again". 
I think we can handle it with the code but is there a easier way to solve this problem? 
Thank you!

**Comment by mpbalshetwar** [comment]
We have handled this using fallback mechanism but I have little different issue. 

My intent is  - I want to book table for tomorrow evening dinner? 

Model is trained and it worked absolutely like gem.

But What if user says - "I don't want to book table for tomorrow evening dinner"

Except word "don't" - everything matches and so confidence is also high and so above intent is picked. How one should handle such cases? 

Quick help would be appreciated.

**Comment by tusharvpatil** [comment]
I have added costume SimplePolicyEnsemble as @Harish0596 but for some reason its not working help me If anything is missing

policy:
`class SimplePolicyEnsemble(PolicyEnsemble):
    def __init__(self, policies, known_slot_events=None):
        super(SimplePolicyEnsemble, self).__init__(policies, known_slot_events)

    def probabilities_using_best_policy(self, tracker, domain):
        result = None
        max_confidence = -1
        for p in self.policies:
            probabilities = p.predict_action_probabilities(tracker, domain)
            confidence = np.max(probabilities)
            if confidence > max_confidence:
                max_confidence = confidence
                result = probabilities

        nlu_confidence = tracker.latest_message.parse_data["intent"]["confidence"]
        if nlu_confidence < 0.5:
            fallback_idx = domain.index_for_action("action_fallback")
            return utils.one_hot(fallback_idx, domain.num_actions)
        elif result and np.max(result) < 0.5:
            # 0.5 is the probability cutoff for rasa core in this case
            # so if the trained policy returned the most likely action with
            # a probability of less than 0.5, we will run the fallback action
            # instead
            fallback_idx = domain.index_for_action("action_fallback")
            return utils.one_hot(fallback_idx, domain.num_actions)
        return result`

custome action:
`class ActionFallback(Action):
    def name(self):
        return "action_fallback"

    def run(self, dispatcher, tracker, domain):
        dispatcher.utter_message("Sorry, didn't get that. Try again.")
        return [UserUtteranceReverted()]`


policies in configuration:
`policies:
  - name: MemoizationPolicy
  - name: TEDPolicy
    max_history: 5
    epochs: 100
  - name: MappingPolicy
  - name: FormPolicy`

**Comment by Vishwa-ud** [comment]
Your input ->  hello                                                                                                              
Hey there! What's your name?
Your input ->  my name is john                                                                                                 
In which city do you live?
Your input ->  im from new york                                                                                                   
Can i get your phone number?
Your input ->  0123456789                                                                                                         
Hey vishwa,New York is a very beautifull place. Your phone number is 0123456789
Your input ->  quit                                                                                                               
Your input ->  what is chat gpt                                                                                                   
I'm here to assist with any questions related to forms. If you need help with our services, features, or how to fill out a form, feel free to ask!
Your input ->  sdas                                                                                                               
Your input ->      

i have some issues  :
1 Out of Context
what is chat gpt or what is world these are out of context messages for these i need to respons like this "I'm here to assist with any questions related to forms. If you need help with our services, features, or how to fill out a form, feel free to ask!" becuse user can ask differnet questions this bot will only focus on form filling. 
2 Unclear
sdas,asdw  and in intent example if i give a something not difine in it there then say  "Sorry, I didn't understand that. Can you rephrase?" ex there is example like this my name is [Jhon] (name) but i give a prompt like this my name is new york. for that i need to say i cant understand can you rephrase like that . and this is not only for name and i need a solution without hardcording how to do this what is the best way use in rasa.
how to get a solution for both senarios.
